infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/LogReplicationAckReader.java

```diff
@@ -43,7 +43,9 @@ public class LogReplicationAckReader {
     private long baseSnapshotTimestamp;
 
     // Periodic Thread which reads the last acknowledged timestamp and writes it to the metadata table
-    private ScheduledExecutorService lastAckedTsPoller;
+    // TODO V2: tune the thread count
+    private static ScheduledExecutorService lastAckedTsPoller = Executors.newScheduledThreadPool(2,
```

Comment:
nit: can you please change 2 to a constant?  It will also be useful to update the comment that the thread pool is shared across all session and fsms